### PR TITLE
multi es-instances support

### PR DESCRIPTION
**Pull request by clarkchen**
I take advantages of Elasticsearch, make host param as a list.

Host Config Change to 
"host1:port1, host2:port " can handle multi es-instances

while  the original port and host can also work well here

### CODE REVIEW

**Comment by clarkchen** [comment]
hello?

File: elastalert/util.py [hunk 1]
```diff
@@ -308,12 +306,17 @@ def build_es_conn_config(conf):
...
-    parsed_conf['es_host'] = os.environ.get('ES_HOST', conf['es_host'])
-    parsed_conf['es_port'] = int(os.environ.get('ES_PORT', conf['es_port']))
     parsed_conf['es_url_prefix'] = ''
     parsed_conf['es_conn_timeout'] = conf.get('es_conn_timeout', 20)
     parsed_conf['send_get_body_as'] = conf.get('es_send_get_body_as', 'GET')
 
+    if "es_host" in conf:
+        es_host = os.environ.get('ES_HOST', conf['es_host'])
+
+        if "es_host" in conf:
```
Comment (Qmando) [inline elastalert/util.py]: es_port

Though, this also isn't necessary.

File: elastalert/util.py [hunk 2]
```diff
@@ -308,12 +306,17 @@ def build_es_conn_config(conf):
     parsed_conf['es_password'] = None
     parsed_conf['aws_region'] = None
     parsed_conf['profile'] = None
-    parsed_conf['es_host'] = os.environ.get('ES_HOST', conf['es_host'])
-    parsed_conf['es_port'] = int(os.environ.get('ES_PORT', conf['es_port']))
     parsed_conf['es_url_prefix'] = ''
     parsed_conf['es_conn_timeout'] = conf.get('es_conn_timeout', 20)
     parsed_conf['send_get_body_as'] = conf.get('es_send_get_body_as', 'GET')
 
+    if "es_host" in conf:
```
Comment (Qmando) [inline elastalert/util.py]: This check isn't necessary, it's always there.

File: elastalert/util.py [hunk 3]
```diff
@@ -344,6 +347,18 @@ def build_es_conn_config(conf):
     return parsed_conf
 
 
+def parse_host(host, port="9200"):
```
Comment (Qmando) [inline elastalert/util.py]: Why not accept a list instead of a comma separated string?

**Comment by YoavBZ** [comment]
Nice! I've just started working on a similar feature :)
In my case, I wanted to run rules on few different hosts and didn't wanted to duplicates my rules, so I added a list of objects to the rule .yaml file that includes the ES hosts, ports and names, so the rule will run separately on each host.
I also appended the environment name to the rule name, i.e "Rule name [Env name]".

How this one is going to work?

**Comment by Qmando** [comment]
I guess the motivation here is that you could replace a load balancer by having a pool of servers to roundrobin requests to.

**Comment by YoavBZ** [comment]
I see.. My idea was more for multiple environments support

**Comment by Qmando** [comment]
So you implemented this feature to avoid duplicating YAML files, is that correct?

You can import bits and pieces of rules, so you could specify two files with just a es_host, name and import parameter.

**Comment by YoavBZ** [comment]
Yes, I wanted to avoid duplicating cause I have one rule with 2/3 ES hosts I want to run it on.
So instead of creating few duplicate rules but with different es_hosts I created one rule with few es_hosts in it

**Comment by YoavBZ** [comment]
*Not one rule actually, I need to run all my rules on few es_hosts

**Comment by sathishdsgithub** [comment]
@YoavBZ @Qmando @clarkchen 

Can we make use of the above modified `util.py` for multi ES instance? I tried with below format but it did not work.
Please, can someone suggest me the right method to mention the list of ES host entry with port?

`es_host: "192.168.96.157:9200, 192.168.96.141:9200"`

**Comment by nsano-rururu** [comment]
Can you resolve conflicts please :)

**Comment by clarkchen** [comment]
> Can you resolve conflicts please :)

sorry, I lost my origin project an start a new pull [request](https://github.com/Yelp/elastalert/pull/3109), where you can review code there

**Comment by clarkchen** [comment]
@sathishdsgithub 

the `ElasticSearchClient` init should change to hosts
image


I add a test case which connect with a docker-composed elastic cluste

you can find the test case [here](https://github.com/clarkchen/elastalert/blob/181103918fb3b684b1b497140aa274d69bfb4a34/tests/util_test.py#L242)

hope it will work

**Pull request closed without merging**